### PR TITLE
Improve install documentation

### LINKED ISSUE #592: SpongeForge installation page provides unclear link to correct Mojang launcher
Currently on the SpongeForge installation steps, there is a warning that using the Minecraft Installer that Mojang provides only installs 8u25, and that Sponge requires 8u40.

It then links to the download page for Minecraft, which effectively links that same installer that causes the issue.

It should either link to an article that explains how to change the Java version for the installed version or link a page which is more clear about what to download.

I've planned on adding this java change as a CMS page on http://hopper.minecraft.net/help/ and will update this issue should it be useful to link to instead, as linking older versions isn't ideal to me.

Page which has the issue: https://github.com/SpongePowered/SpongeDocs/blob/stable/source/server/getting-started/implementations/spongeforge.rst#installing-spongeforge-1 (https://docs.spongepowered.org/stable/en/server/getting-started/implementations/spongeforge.html)

### PR DESCRIPTION
**Pull request by ST-DDT**
Fixes #592

### CODE REVIEW

**Comment by Spongy** [comment]
Setting up PR reference, please wait...

**Comment by phit** [comment]
mh while you are at this, I wrote this a while ago because people keep trying to match forge and sponge exactly https://gist.github.com/phit/d097a5a4e7172e6c1e4a7616596bc7ce

File: source/server/getting-started/jre.rst [hunk 1]
```diff
@@ -19,11 +19,16 @@ Java, and it should work just as well, if not better. However, it's also possibl
...
+
+* launcher using ``Profile`` (``Advanced Options``) -> ``<Your Profile>`` -> ``Java-Executable`` (``.../java.exe``)
+* server using either the ``PATH`` variable or using the full path to your java executable
+
 32-bit vs. 64-bit
 ~~~~~~~~~~~~~~~~~
 
 If your computer supports it, you should use 64-bit versions of Java whenever possible. The Java installers from the
-linked website should detect whether your computer is ready for 64-bit.
+`this website <https://java.com/en/download/>`__ should detect whether your computer is ready for 64-bit.
```
Comment (phit) [inline source/server/getting-started/jre.rst]: probably wiser to use https://java.com/en/download/manual.jsp

Comment (ST-DDT) [reply source/server/getting-started/jre.rst]: Not sure about that, there is the selection, but it does not really help you with selecting 32 or 64 bit.

**Comment by ST-DDT** [comment]
@phit Like this? https://github.com/SpongePowered/SpongeDocs/pull/639/commits/3b52925c9a36236e50470e088877cf232255f101

**Comment by phit** [comment]
yeah, feel free to proofread and improve on that, I just wanted to make it clearer that you don't always have to use the exact version and avoid people asking when SpongeForge will get updated to Forge XXXX

File: source/server/getting-started/implementations/spongeforge.rst [hunk 2]
```diff
@@ -23,33 +23,45 @@ However, SpongeForge usually updates to a new Forge build fairly soon after it's
...
 
-    Normal Forge mods can usually run on any build of Forge for a given Minecraft version (e.g. 1.8.0) without
-    any problems. However, SpongeForge needs to access, among other things, internal parts of Forge, which
-    most mods shouldn't be touching, let alone modifying as Sponge does. Since Forge is free to change internal
-    code whenever they want to, its normal guarantee of backwards-compatibility doesn't apply to SpongeForge.
+    Normal Forge mods can usually run on any build of Forge for a given Minecraft version (e.g. 1.8.0) without any
+    problem. However, SpongeForge needs to access, among other things, internal parts of Forge, which most mods
+    shouldn’t be touching, let alone modifying as Sponge does. Since Forge is free to change their internal code
+    whenever they want to, its normal guarantee of backwards-compatibility doesn’t apply to SpongeForge. Feel free to
+    use more recent versions of Forge, than the one used for SpongeForge, but we can't always garantee compatibility.
```
Comment (ryantheleach) [inline source/server/getting-started/implementations/spongeforge.rst]: guarantee

File: source/server/getting-started/implementations/spongeforge.rst [hunk 3]
```diff
@@ -23,33 +23,45 @@ However, SpongeForge usually updates to a new Forge build fairly soon after it's
...
+
 
+.. warning::
+    
+    When investigating crash issues, you can freely use a newer versions of Forge than listed on the SpongeForge Jar.
+    However it is recommended to also check with the matching versions, to make sure your issue is not related to the
+    version mismatch. 
+    Even though there will be no guarantee of compatibility, please report any breakage to the issue tracker, so that
+    SpongeForge can be updated.
```
Comment (ryantheleach) [inline source/server/getting-started/implementations/spongeforge.rst]: Maybe we should provide an example of a failing mixin due to forge bytecode changing?

File: source/server/getting-started/implementations/spongeforge.rst (hunk 2 above)
Comment (phit) [reply source/server/getting-started/implementations/spongeforge.rst]: blame me

File: source/server/getting-started/implementations/spongeforge.rst [hunk 4]
```diff
@@ -23,33 +23,45 @@ However, SpongeForge usually updates to a new Forge build fairly soon after it's
...
-| ``SpongeBuildId``    | The build number of Sponge. This is what you should supply when reporting bugs or seeking    |
-|                      | support.                                                                                     |
-+----------------------+----------------------------------------------------------------------------------------------+
+The format of the filename is ``spongeforge-<MCVersion>-<ForgeBuildId>-<SpongeAPIVersion>-BETA-<SpongeBuildId>.jar``
+
++----------------------+-----------------------------------------------------------------------------------------------+
+| ``MCVersion``        | The Minecraft version. Only clients compatible with this version can connect.                 |
++----------------------+-----------------------------------------------------------------------------------------------+
+| ``ForgeBuildId``     | The build number of Forge this file is built for. Preferably your server should run this exact|
+|                      | version of Forge. Which is the last part of Forge's version string.                           |
```
Comment (Inscrutable) [inline source/server/getting-started/implementations/spongeforge.rst]: This should be one sentence, and made a bit clearer. ie. ``Preferably your server should run this exact version of Forge (which can be identified in the last part of Forge's version string)``.

File: source/server/getting-started/implementations/spongeforge.rst [hunk 5]
```diff
@@ -23,33 +23,45 @@ However, SpongeForge usually updates to a new Forge build fairly soon after it's
...
+    Normal Forge mods can usually run on any build of Forge for a given Minecraft version (e.g. 1.8.0) without any
+    problem. However, SpongeForge needs to access, among other things, internal parts of Forge, which most mods
+    shouldn’t be touching, let alone modifying as Sponge does. Since Forge is free to change their internal code
+    whenever they want to, its normal guarantee of backwards-compatibility doesn’t apply to SpongeForge. Feel free to
+    use more recent versions of Forge, than the one used for SpongeForge, but we can't always garantee compatibility.
+
 
+.. warning::
+    
+    When investigating crash issues, you can freely use a newer versions of Forge than listed on the SpongeForge Jar.
```
Comment (Inscrutable) [inline source/server/getting-started/implementations/spongeforge.rst]: Some plural problems here. The "a" is not needed in this sentence. However, the next sentence should refer to a singular matching *version*, and I'd refer to "a version mismatch" (there may not be one).

**Comment by ST-DDT** [comment]
* [x] Mention Sponge needs Java 8.0.40
* [x] Link to how to install Java / config JRE
* [ ] Mention alternative download section
* [x] ~~Automatic download page~~

**Comment by ST-DDT** [comment]
@Inscrutable In https://github.com/SpongePowered/SpongeDocs/issues/592#issuecomment-294672757 you mentioned an alternative minecraft download page for a version without Java embedded. Are you referring to that page: https://minecraft.net/en-us/download/alternative ?

**Comment by kinggoesgaming** [comment]
Mention that sponge only work with java 8 not 9

**Comment by ST-DDT** [comment]
I did that [here](https://github.com/SpongePowered/SpongeDocs/pull/639/files#diff-2da373ae50fb66de4e5d1653a6b6b26fR9) or do you want it elsewhere as well?

**Comment by ST-DDT** [comment]
Ready for review.

File: source/server/getting-started/implementations/spongevanilla.rst [hunk 6]
```diff
@@ -30,10 +46,11 @@ SpongeVanilla only works as a dedicated server.
 
 .. warning::
 
-    Don't double-click the ``.jar`` file!
+    Don't double-click the ``.jar`` file! As it will start the server without a window to control it.
```
Comment (Inscrutable) [inline source/server/getting-started/implementations/spongevanilla.rst]: This should be one sentence.

File: source/server/getting-started/jre.rst (hunk 1 above)
Comment (Inscrutable) [reply source/server/getting-started/jre.rst]: Some browsers can handle the auto-detection, and some (eg. *Pale Moon*) cannot. The manual downloads page is a good place to make sure you *get* a 64-bit version, but you can also have multiple versions of Java installed so it's no guarantee that it'll be the one used by software. The Java control panel will give you the info in Windows, but I'm not sure about Mac or Linux systems.

**Pull request merged**